objects/src/lib.rs

```diff
@@ -33,6 +33,10 @@ pub mod assets;
 pub use assets::{Asset, FungibleAsset, NonFungibleAsset, NonFungibleAssetDetails};
 
 pub mod notes;
+pub use notes::{
+    Note, NoteEnvelope, NoteInclusionProof, NoteInputs, NoteMetadata, NoteOrigin, NoteScript,
+    NoteStub, NoteVault,
+};
```

Comment:
I'm not sure if we should do this. Two considerations here:

1. If we are re-exporting all individual structs, do we also need to make the entire module public as well? Doing either one or the other seems more appropriate.
2. If we re-export internal structs from all sub-module, library interface will be quite "polluted". So, maybe instead we should keep the modules public but not re-export internal structs publicly.

The second point would imply that we'd change this to something like:
```Rust
pub mod notes;
use notes::{
    Note, NoteEnvelope, NoteInclusionProof, NoteInputs, NoteMetadata, NoteOrigin, NoteScript,
    NoteStub, NoteVault,
};
```
And would make similar changes to `assets` and `accounts` modules above.

Reply:
I'm unsure what is the request here. I'm happy to change the style, I'm just doing exports because this seems to be the practice used through out existing crates.

To make reviewing and contributing to our repos easier, it would be super useful to  codify / write down all matters of style into a `CONTRIBUTING.md` file. That way we have a reference for ourselves, and ideally wouldn't need to have these discussions on PR reviews, speeding up our reviewing process and also making external contributions easier.

For this particular case, my motivation was just to keep it consistent with the existing exports for the `accounts`, and as mentioned before, this seems to be common practice on the other crates too. My understanding is that these exports are done by an heuristic, were useful types are re-exported, this happened to be useful for the builder pattern.

If it helps, my opinion is that we should not use heuristics for these things, since that always leaves room for interpretation, I would only use exports to keep backwards compatibility, and when there is a need for convenience my understanding is that the common practice is to have `prelude` module for that.

Reply:
> I'm unsure which of the two options is the preferred one, removed the re-exports. I'm happy to change the style, I'm just doing exports because this seems to be the practice used through out existing crates.
>
> To make reviewing and contributing to our repos easier, it would be super useful to codify / write down all matters of style into a CONTRIBUTING.md file. That way we have a reference for ourselves, and ideally wouldn't need to have these discussions on PR reviews, speeding up our reviewing process and also making external contributions easier.

Yeah, I'm unsure myself on what's the best approach here. Generally, I tend to bring everything up to the root level unless the number of structs/enums etc. becomes big. And then I put them into logically-grouped submodules, but don't export the module itself.

So, one rule we can have is that if we re-export everything from some sub-module publicly, there is no need to make the submodule itself public (to apply this consistently, we'd need to change how `assets` are exported).

> For this particular case, my motivation was just to keep it consistent with the existing exports for the `accounts`, and as mentioned before, this seems to be common practice on the other crates too. My understanding is that these exports are done by an heuristic, were useful types are re-exported, this happened to be useful for the builder pattern.

Yep, makes sense. I'm just unsure if now the number of re-exported structs at the root level is too big and we should change how we do accounts and asset exports (to just export modules).

> If it helps, my opinion is that we should not use heuristics for these things, since that always leaves room for interpretation, I would only use exports to keep backwards compatibility, and when there is a need for convenience my understanding is that the common practice is to have `prelude` module for that.

Generally agree, but we haven't been using `prelude` anywhere else, and not sure if it makes sense to introduce it now. But I do think that we should re-visit external interface of this create more holistically separately.